src/mame/konami/hexion.cpp

```diff
@@ -231,12 +242,21 @@ void hexion_state::bankswitch_w(uint8_t data)
 	// bits 0-3 select ROM bank
 	m_rombank->set_entry(data & 0x0f);
 
-	// does bit 6 trigger the 052591?
+	// bit 6 triggers the 052591
 	if (data & 0x40)
 	{
-		int bank = m_unkram[0] & 1;
-		memset(m_vram[bank], m_unkram[1], 0x2000);
-		m_bg_tilemap[bank]->mark_all_dirty();
+		uint8_t command = m_pmcram[0];
+		if (command <= 1)
+		{
+			memset(m_vram[command], m_pmcram[1], 0x2000);
+			m_bg_tilemap[command]->mark_all_dirty();
+		}
+		else
+		{
+			uint8_t bank = m_pmcram[4] & 1;
+			memset(m_vram[bank] + (get_u16le(&m_pmcram[2]) & 0x1fff), 0, 16);
+			m_bg_tilemap[bank]->mark_all_dirty();
```

Comment:
There’s still a potential heap smash here – consider what happens if `(get_u16le(&m_pmcram[2]) & 0x1fff)` is 0x1ff8.

Reply:
Fixed on my side. Should I submit a new pull request ?

Reply:
I already fixed it.